File: desktop/src/main/java/com/corrodinggames/rts/gameFramework/graphics/GamePaint.java
package com.corrodinggames.rts.gameFramework.graphics;

import android.graphics.Paint;
import android.graphics.Typeface;
import com.corrodinggames.rts.gameFramework.GameEngine;

/* JADX INFO: renamed from: com.corrodinggames.rts.gameFramework.m.ag */
/* JADX INFO: loaded from: game-lib.jar:com/corrodinggames/rts/gameFramework/m/ag.class */
public class GamePaint extends Paint {
    public static final GamePaint r = new GamePaint();
    ShaderProgram t;
    boolean s = false;
    boolean u = false;

    static {
        r.b(-1);
        r.o();
    }

    public void o() {
        this.u = true;
    }

    public void c(float f) {
        super.b(f);
    }

    @Override // android.graphics.Paint
    public void b(float f) {
        if (this.u) {
            GameEngine.logColored("UniquePaint changed when locked down:");
            GameEngine.logColored("from:" + k() + " to: " + f);
            GameEngine.printStackTrace();
        }
        super.b(f);
    }

    @Override // android.graphics.Paint
    public Typeface a(Typeface typeface) {
        if (this.u) {
            GameEngine.logColored("UniquePaint changed when locked down:");
            GameEngine.printStackTrace();
        }
        return super.a(typeface);
    }

    public static void b(Paint paint) {
        ((GamePaint) paint).o();
    }

    public boolean p() {
        return this.s;
    }

    @Override // android.graphics.Paint
    public void a(boolean z) {
        this.s = z;
        super.a(z);
    }

    public ShaderProgram q() {
        return this.t;
    }

    public void a(ShaderProgram shaderProgram) {
        this.t = shaderProgram;
    }
}

File: core/src/main/kotlin/com/corrodinggames/rts/gameFramework/graphics/GamePaint.kt
package com.corrodinggames.rts.gameFramework.graphics

import com.corrodinggames.rts.gameFramework.GameEngine
import io.github.rwx.render.canvas.KoolDisplacementEffect
import io.github.rwx.render.canvas.KoolPaint
import io.github.rwx.render.canvas.KoolTypeface

class GamePaint : KoolPaint() {
    private var displacementEffect: KoolDisplacementEffect? = null
    private var shaderProgram: ShaderProgram? = null
    private var hasFlag: Boolean = false
    private var locked: Boolean = false

    fun o() {
        locked = true
    }

    fun c(value: Float) {
        super.b(value)
    }

    override fun b(value: Float) {
        if (locked) {
            GameEngine.logColored("UniquePaint changed when locked down:")
            GameEngine.logColored("from:" + textSize() + " to: " + value)
            GameEngine.printStackTrace()
        }
        super.b(value)
    }

    override fun a(typeface: KoolTypeface?): KoolTypeface? {
        if (locked) {
            GameEngine.logColored("UniquePaint changed when locked down:")
            GameEngine.printStackTrace()
        }
        return super.a(typeface)
    }

    fun p(): Boolean = hasFlag

    fun hasLegacyFlag(): Boolean = p()

    override fun a(value: Boolean) {
        hasFlag = value
        super.a(value)
    }

    fun q(): KoolDisplacementEffect? = displacementEffect

    fun displacementEffect(): KoolDisplacementEffect? = q()

    fun a(effect: KoolDisplacementEffect?) {
        if (displacementEffect !== effect) {
            displacementEffect = effect
            markBackendStateChanged()
        }
    }

    fun setDisplacementEffect(effect: KoolDisplacementEffect?) = a(effect)

    fun shaderProgram(): ShaderProgram? = shaderProgram

    fun a(shaderProgram: ShaderProgram?) {
        if (this.shaderProgram !== shaderProgram) {
            this.shaderProgram = shaderProgram
            markBackendStateChanged()
        }
    }

    fun setShaderProgram(shaderProgram: ShaderProgram?) = a(shaderProgram)

    companion object {
        @JvmField
        val r: GamePaint = GamePaint().apply {
            setColor(-1)
            o()
        }

        @JvmStatic
        fun b(paint: KoolPaint) {
            (paint as GamePaint).o()
        }
    }
}
